src/wxflow/scheduler/scheduler.py

```diff
@@ -0,0 +1,209 @@
+
+        *args : Any
+            Additional positional arguments.
+        **kwargs : Any
+            Additional keyword arguments.
+
+        """
+
+        # Cache incoming config
+        self._config = AttrDict(config)
+
+        self._config_to_specs()
+        self.batch_card = []
+
+        # Set attributes from the user provided information
+        for arg in args:
+            setattr(self, str(arg), arg)
+
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def _config_to_specs(self) -> None:
+        """
+        Deep copies and converts the internal configuration dictionary to a standardized specification format.
+        This method processes the internal `_config` attribute and assigns the self.specs attribute with the following transformations:
+          - Converts the 'memory' field to a string in megabytes (e.g., '1024M').
+          - Converts the 'walltime' field to a string in HH:MM:SS format.
+          - Ensures the 'env' field (environment variables) is a list.
+          - Ensures the 'native' field (native scheduler directives) is a list.
+        """
+        self.specs = self._config.deepcopy()
+
+        # Convert 'memory' to MB
+        if 'memory' in self._config:
+            self.specs.memory = str(self.memory_in_megabytes(self._config.memory)) + 'M'
+
+        # Convert 'walltime' to HH:MM:SS format
+        if 'walltime' in self._config:
+            self.specs.walltime = self.walltime_in_string(self._config.walltime)
+
+        # Ensure environment variables are a list
+        if 'env' in self._config:
+            if not isinstance(self._config.env, (list, tuple)):
+                self.specs.env = [self._config.env]
+
+        # Ensure native scheduler directives are a list
+        if 'native' in self._config:
+            if not isinstance(self._config.native, (list, tuple)):
+                self.specs.native = [self._config.native]
+
+        return
+
+    def dump(self, filename: Optional[str] = None) -> None:
+        """
+        Dumps the batch card to a file or prints it to stdout.
+
+        If a filename is provided, writes the contents of `self.batch_card` to the specified file.
+        Otherwise, prints the batch card to standard output.
+
+        Parameters
+        ----------
+        filename : Optional[str], default=None
+            The path to the file where the batch card should be written. If None, the batch card is printed to stdout.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        Exception
+            If an error occurs while writing to the specified file.
+        """
+        if filename is not None:
+            try:
+                with open(filename, 'w') as fh:
+                    for item in self.batch_card:
+                        fh.write(f"{item}\n")
+            except Exception as e:
+                raise f"Unknown exception in writing scheduler directives to {filename} from {e}"
```

Comment:
Exceptions cannot be converted directly to strings.  Instead, use `type(e).__name__`.
```suggestion
                raise RuntimeError(f"Encountered a {type(e).__name__} while writing scheduler directives to {filename}")
```

Reply:
does `str(e)` not work?

Reply:
That would work as well.  I didn't know that!  Thanks.

Reply:
I see now that exceptions automatically get converted to strings.  I seem to recall this wasn't true in the past, but I may be wrong.  I think the only issue then is that `raise` needs to be flowed by an `Exception` class (e.g. `raise Exception("...")`.

Reply:
I updated this line to `raise Exception(message)`